content/en/user-guide/aws/s3/index.md

```diff
@@ -45,58 +99,66 @@ $ awslocal s3api put-object --bucket sample-bucket --key index.html --body index
 }
 {{< / command >}}
 
+### Generate a pre-signed URL for S3 object
 
-{{< alert title="Note" >}}
-Just like AWS, LocalStack differentiates between [Path-Style and Virtual Hosted-Style Requests](https://docs.aws.amazon.com/AmazonS3/latest/userguide/VirtualHosting.html) depending on your `Host` header for a request.
+You can generate a pre-signed URL for your S3 object using the [`presign`](https://docs.aws.amazon.com/cli/latest/reference/s3/presign.html) command. Pre-signed URL allows anyone to retrieve the S3 object with an HTTP GET request.
+
+Run the following command to generate a pre-signed URL for your S3 object:
+
+{{< command >}}
+$ awslocal s3 presign s3://sample-bucket/image.jpg
+{{< / command >}}
+
+You will see a generated pre-signed URL for your S3 object. You can use [`cURL`](https://curl.se/) or [`wget`](https://www.gnu.org/software/wget/) to retrieve the S3 object using the pre-signed URL.
 
-Example:
+## Path-Style and Virtual Hosted-Style Requests
 
-```plaintext
+Similar to AWS, LocalStack categorizes requests as either [Path-Style or Virtual Hosted-Style](https://docs.aws.amazon.com/AmazonS3/latest/userguide/VirtualHosting.html) based on the Host header of the request. The following example illustrates this distinction:
+
+```bash
 <bucket-name>.s3.<region>.localhost.localstack.cloud # host-style request
 ```
 
 As a special case in LocalStack, leaving out `<region>` also works for the `s3.localhost.localstack.cloud` domain:
 
-`<bucket-name>.s3.localhost.localstack.cloud` is also a host-style request.
-
-All other requests will be considered path-style requests. It is advised to use the `s3.localhost.localstack.cloud` endpoint URL for all requests targeting S3.
-{{% /alert %}}
+As a special case in LocalStack, leaving out `<region>` also works for the `s3.localhost.localstack.cloud` domain:
 
-## S3 Providers
+In LocalStack, a special case exists where omitting the `<region>` also functions correctly for the `s3.localhost.localstack.cloud domain`. 
 
-LocalStack's S3 support is currently available via two providers: `legacy` and `v2` (formerly known as `asf`). For users, switching between the two providers has a lot of impacts. Using the `PROVIDER_OVERRIDE_S3`, you can switch between the two providers. The `legacy` provider is the default provider and the `v2` provider is our new and more stable provider. The `legacy` provider is loaded by default, and you need to set `PROVIDER_OVERRIDE_S3=v2` to use the new provider.
+`<bucket-name>.s3.localhost.localstack.cloud` is also a host-style request.
 
-With v2.0, the default will be changed to `v2`, but the legacy provider will still be available (using the feature flag with the value `legacy`), though it will be removed in further releases.
+All other requests are treated as path-style requests. Using the `s3.localhost.localstack.cloud` endpoint URL is recommended for all requests aimed at S3.
 
 ## Storage Configuration
```

Comment:
I would like to remove this part entirely for now. S3_DIR is not well supported, and it will wreck LocalStack in weird manners. It does not support multiparts uploads, and lambda now relies on them, which can make everything fall apart. Not many people use S3_DIR, and it would be nice if not more people would start using it now. I'm currently in the process of rewriting the whole logic behind it and then it would be transparent and kinda work like persistence, but for arbitrary mounted directories.
But in the meantime, it would be great if we wouldn't be advertising it. I know it feels bad, but we should keep this once the feature is ready! Does that make sense?